src/v/cluster/partition_leaders_table.h

```diff
@@ -86,6 +91,7 @@ class partition_leaders_table {
     void update_partition_leader(
       const model::ntp&,
       model::revision_id,
```

Comment:
Do we need both revision_id and initial_revision_id fields? Perhaps letting health monitor and metadata dissemination service choose which revision to use for the update will be less invasive?